tests/10apidoc/01register.pl

```diff
@@ -126,7 +128,75 @@ sub matrix_register_user
    });
 }
 
-push @EXPORT, qw( local_user_fixture local_user_fixtures );
+push @EXPORT, qw( matrix_register_user_via_secret );
+
+sub matrix_register_user_via_secret
+{
+   my ( $http, $uid, %opts ) = @_;
+
+   my $password = $opts{password} // "an0th3r s3kr1t";
+   my $admin = $opts{admin} // 0;
+
+   defined $uid or
+      croak "Require UID for matrix_register_user_via_secret";
+
+   my $hmac = Digest::HMAC_SHA1->new("reg_secret");
```

Comment:
Hrm. This shared-secret has to be the same value as the one given at synapse configuration time? This seems a bit fragile to just keep two string literals that must have equal value - maybe we can store it somewhere?